packages/babel-plugin-apply-mdx-type-props/index.js

```diff
@@ -2,28 +2,41 @@ const {types: t} = require('@babel/core')
 const {declare} = require('@babel/helper-plugin-utils')
 const {startsWithCapitalLetter} = require('@mdx-js/util')
 
+const applyMdxType = (node, jsxName) => {
+  node.attributes.push(
+    t.jSXAttribute(t.jSXIdentifier(`mdxType`), t.stringLiteral(jsxName))
+  )
+}
+
 class BabelPluginApplyMdxTypeProp {
   constructor() {
     const names = []
     this.state = {names}
 
     this.plugin = declare(api => {
       api.assertVersion(7)
+      const {types: t} = api
 
       return {
         visitor: {
           JSXOpeningElement(path) {
+            if (t.isJSXMemberExpression(path.node.name)) {
+              const objName = path.node.name.object.name
+              const propName = path.node.name.property.name
+              const fullJsxName = `${objName}.${propName}`
```

Comment:
this would also need to be checked if we swapped to something more full featured in the `getMemberExpression` call. Perhaps we should add a warning if there are too many nested objects.